### PR TITLE
AP_Baro: KellerLD: clean up reading of calibration data

### PR DESCRIPTION
**Pull request by peterbarker**
Co-authored-by: Josh Henderson <hendjoshsr71@gmail.com>

Checks all return values from transfer functions to avoid use of
uninitialised data.

### CODE REVIEW

**Comment by peterbarker** [comment]
... also saves ~200 bytes on CubeBlack for a Sub build

**Comment by hendjoshsr71** [comment]
LGTM

**Comment by jaxxzer** [comment]
@hendjoshsr71 
@patrickelectric have one of you got one of these sensors on hand that you are testing with?

**Comment by hendjoshsr71** [comment]
Rebased this one though https://github.com/ArduPilot/ardupilot/pull/18983 will conflict and should go first.

**Review by Williangalvani — APPROVED** [review]
Just tested on a Bar100 (PA-7LD) and it works fine :+1:

**Comment by hendjoshsr71** [comment]
Rebased for hopefully the last time. 

@Williangalvani you may wish to take another look?

File: libraries/AP_Baro/AP_Baro_KellerLD.cpp [hunk 1]
```diff
@@ -61,37 +61,91 @@ AP_Baro_Backend *AP_Baro_KellerLD::probe(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::D
...
+}
+
+// Read sensor P-Mode type and set pressure reference offset
+// This determines the pressure offset based on the type of sensor
+// vented to atmosphere, gauged to vacuum, or gauged to standard sea-level pressure
+bool AP_Baro_KellerLD::read_mode_type()
+{
+    uint8_t cmd = CMD_METADATA_PMODE;
+    uint8_t data[3];
+    if (!transfer_with_delays(&cmd, 1, &data[0], 3)) {
```
Comment (peterbarker) [inline libraries/AP_Baro/AP_Baro_KellerLD.cpp:140]: Pre-existing, but should probably be `ARRAY_SIZE(data)`.  And just pass in data....

**Review by peterbarker — COMMENTED** [review]
I think we just need the nullptr check back..

And assurance that the thing still works :-)

File: libraries/AP_Baro/AP_Baro_KellerLD.cpp [hunk 2]
```diff
@@ -61,37 +61,91 @@ AP_Baro_Backend *AP_Baro_KellerLD::probe(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::D
...
 
-// The hardware does not need to be reset/initialized
-// We read out the measurement range to be used in raw value conversions
-bool AP_Baro_KellerLD::_init()
+
+// convenience function to work around device transfer oddities
+bool AP_Baro_KellerLD::transfer_with_delays(uint8_t *send, uint8_t sendlen, uint8_t *recv, uint8_t recvlen)
 {
-    if (!_dev) {
+    if (!_dev->transfer(send, sendlen, nullptr, 0)) {
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply libraries/AP_Baro/AP_Baro_KellerLD.cpp:68]: Fixed. what I get for not opening up the files outside of git when the `diff` is botched so horribly.

File: libraries/AP_Baro/AP_Baro_KellerLD.cpp (hunk 1 above)
Comment (hendjoshsr71) [reply libraries/AP_Baro/AP_Baro_KellerLD.cpp]: Done

**Comment by hendjoshsr71** [comment]
@Williangalvani If you wouldn't mind giving this another look over and test again that would be appreciated!

**Comment by Williangalvani** [comment]
Working fine. I think the CI failure is unrelated

**Comment by hendjoshsr71** [comment]
Tests are passing now, after I kicked them off again.

**Pull request merged**